api.js

```diff
@@ -108,6 +109,15 @@ Api.prototype._handleExceptions = function (data) {
 };
 
 Api.prototype._handleStats = function (stats) {
+	if (this.hasExclusive && !stats.hasExclusive) {
+		return;
+	}
+
+	if (!this.hasExclusive && stats.hasExclusive) {
+		this.hasExclusive = true;
+		this.testCount = 0;
```

Comment:
Similarly why is this being reset? Because we've already counted tests we're now not going to run, right? I appreciate `test-collection` has similar logic without comments but I reckon it'd be easier to understand for the next person if we'd explain it here.